### PR TITLE
Fix/adding https enable flag

### PR DESCRIPTION
**Pull request by saumilmac**
What:
- Adding a flag to enable HTTPS
- Adding snapshot to test deployment when https is enabled

### CODE REVIEW

File: charts/snyk-broker/values.yaml [hunk 1]
```diff
@@ -192,6 +192,9 @@ logEnableBody: "false"
 
 ##### Enable HTTPS #####
 
+# To enable broker client to run a HTTPS server enable enableBrokerServerOverHttps flag and also provide location of HTTPS_CERT and HTTPS_KEY
+enableBrokerServerOverHttps: false
```
Comment (soniqua) [inline charts/snyk-broker/values.yaml:196]:
```suggestion
forceHttps: false
```

File: charts/snyk-broker/templates/broker_deployment.yaml [hunk 2]
```diff
@@ -58,7 +58,7 @@ spec:
             httpGet:
               port: {{ .Values.deployment.container.containerPort }}
               path: {{ .Values.brokerLivenessProbe.path }}
-              {{- if and (.Values.httpsCert) (.Values.httpsKey) }}
+              {{- if .Values.enableBrokerServerOverHttps }}
```
Comment (soniqua) [inline charts/snyk-broker/templates/broker_deployment.yaml:61]: This changes existing behaviour - we should maintain backwards compatibility:

```suggestion
              {{- if or ( and (.Values.httpsCert) (.Values.httpsKey) ) ( .Values.enableBrokerServerOverHttps ) ) }}
```

**Review by soniqua — CHANGES_REQUESTED** [review]

File: charts/snyk-broker/values.yaml (hunk 1 above)
Comment (soniqua) [reply charts/snyk-broker/values.yaml:196]: We should consider adding a snapshot for this too.

Comment (saumilmac) [reply charts/snyk-broker/values.yaml:196]: I can add a snapshot for this. @aarlaud any thoughts on the name here ?

Comment (aarlaud) [reply charts/snyk-broker/values.yaml:196]: It was meant to be BrokerServe not BrokerServer. It's for the local broker server running in the client. Referencing broker server is confusing here, but broker client, which most people refer to broker, then does server over https.
Ideally it should be enableBrokerLocalWebserverOverHttps. Maybe we just use that long name so it's clearer?

File: charts/snyk-broker/templates/broker_deployment.yaml (hunk 2 above)
Comment (saumilmac) [reply charts/snyk-broker/templates/broker_deployment.yaml:61]: @soniqua I see your point but my thought process here was keep a single flag to choose between https vs http. If httpsEnabled flag is set httpsCert and httpsKey are mandatory.

File: charts/snyk-broker/values.yaml (hunk 1 above)
Comment (soniqua) [reply charts/snyk-broker/values.yaml:196]: A longer name is not an issue. The main problem was `server` as opposed to `serve`.

File: charts/snyk-broker/templates/broker_deployment.yaml (hunk 2 above)
Comment (saumilmac) [reply charts/snyk-broker/templates/broker_deployment.yaml:61]: ahh from that lens, yes extra value will be required. I see your point now, I ll add the double conditional. But for the new hybrid platform chart we should keep a single flag to enable/disable something.

File: charts/snyk-broker/values.yaml (hunk 1 above)
Comment (saumilmac) [reply charts/snyk-broker/values.yaml]: fixed

**Review by aarlaud — APPROVED** [review]

**Review by soniqua — APPROVED** [review]

**Pull request merged**